modules/preview-react/menu/stories/examples/Basic.tsx

```diff
@@ -1,17 +1,74 @@
 import React from 'react';
-
 import {Menu, MenuItem} from '@workday/canvas-kit-preview-react/menu';
+import {SecondaryButton} from '@workday/canvas-kit-react/button';
+import {
+  Popup,
+  usePopupModel,
+  useAlwaysCloseOnOutsideClick,
+  useReturnFocus,
+  useCloseOnEscape,
+} from '@workday/canvas-kit-react/popup';
+
+const menuId = 'basic-menu';
 
 export const Basic = () => {
+  const model = usePopupModel();
+
+  useAlwaysCloseOnOutsideClick(model);
+  useCloseOnEscape(model);
+  useReturnFocus(model);
+
+  const isOpen = model.state.visibility !== 'hidden';
+
+  const handleButtonKeyDown = (event: React.KeyboardEvent) => {
+    let isShortcut = false;
+    if (event.key === `Spacebar` || event.key === ` ` || event.key === `Enter`) {
+      isShortcut = true;
+      if (isOpen) {
+        model.events.hide();
+      }
+      model.events.show();
```

Comment:
Is it intended that when `isOpen` is true, it will hide then show? Or should it be an else clause wrapping this?

Reply:
If it is intended, `models.events.show()` could go inside the `if (isShortcut)` block.

Reply:
No, there should be an `else` wrapping the `model.events.show()`. Good catch 👍